File: 2015/day01-1.hs
import System.IO

solution :: String -> Int
solution content = (countChar '(' content) - (countChar ')' content)

countChar :: Eq a => a -> [a] -> Int
countChar _ [] = 0
countChar ch (x:xs)
    | ch == x = 1 + countChar ch xs
    | otherwise = countChar ch xs

main :: IO ()
main = do
    handle <- openFile "day01.dat" ReadMode
    content <- hGetContents handle
    print . show . solution $ content

File: 2015/day01-1.ts
function solution(content: string) {
  console.log(
    content.split("").filter((char) => char === "(").length -
      content.split("").filter((char) => char === ")").length
  );
}

Deno.readTextFile("day01.dat")
  .then((content) => content.trim())
  .then(solution);
